packages/xdg_directories/example/lib/main.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'dart:io';
+import 'package:flutter/material.dart';
+import 'package:xdg_directories/xdg_directories.dart';
+import 'widgets/directory_display.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  const MyApp({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'XDG Directories Demo',
+      theme: ThemeData(
+        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
+        useMaterial3: true,
+      ),
+      home: const MyHomePage(title: 'XDG Directories Demo'),
+    );
+  }
+}
+
+class MyHomePage extends StatefulWidget {
+  const MyHomePage({super.key, required this.title});
+
+  final String title;
+
+  @override
+  State<MyHomePage> createState() => _MyHomePageState();
+}
+
+class _MyHomePageState extends State<MyHomePage> {
+  final Set<String> userDirectoryNames = getUserDirectoryNames();
+  String selectedUserDirectory = '';
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
+        title: Text(widget.title),
+      ),
+      body: Center(
+        child: ListView(
+          shrinkWrap: true,
+          children: <Widget>[
+            const Padding(
+              padding: EdgeInsets.all(10.0),
+              child: Text(
+                'Select a user directory name:',
+                style: TextStyle(
+                  color: Colors.black87,
+                  fontSize: 25,
+                ),
+                textAlign: TextAlign.start,
+              ),
+            ),
+            ListView.builder(
+              shrinkWrap: true,
+              itemCount: userDirectoryNames.length,
+              itemBuilder: (BuildContext context, int index) => ListTile(
+                title: Text(userDirectoryNames.elementAt(index)),
+                visualDensity: VisualDensity.standard,
+                onTap: () => setState(
+                  () {
+                    selectedUserDirectory =
```

Comment:
Requiring a button tap and only showing a single value at a time make this example much harder to use to validate functionality than it needs to be; it can simply be a list of directory types and their values, with no interaction at all.

The purpose of the example app isn't to show off Flutter functionality in general, just demonstrate the specific package's API.

Reply:
Removed the list tiles and created a fixed list with type/values.